kernel/kernel-util/src/main/java/org/sakaiproject/util/EmailNotification.java

```diff
@@ -645,6 +645,24 @@ protected int getOption(User user, String notificationId, String resourceFilter,
 		String type = getType(resourceFilter);
 		if (type != null)
 		{
+            // First, check the overrides.
+            String eventContext = event.getContext();
+            if (eventContext != null) {
+                props = prefs.getProperties(NotificationService.PREFS_TYPE + type + NotificationService.NOTI_OVERRIDE_EXTENSION);
+                Iterator<String> i = props.getPropertyNames();
+                while (i.hasNext()) {
+                    String name = i.next();
+                    if (eventContext.equals(name)) {
+                        try {
+                            int option = Integer.parseInt(props.getProperty(name));
+                            if (option != NotificationService.PREF_NONE) return option;
+                        } catch (NumberFormatException nfe) {
+                            nfe.printStackTrace();
```

Comment:
Having errors that don't go through the logging framework can make them get lost. Most of the other code looks to just swallow exceptions or return a default value.